akka-http/src/main/scala/akka/http/scaladsl/server/directives/FileAndResourceDirectives.scala

```diff
@@ -222,7 +222,16 @@ object FileAndResourceDirectives extends FileAndResourceDirectives {
             ResourceFile(url, entry.getSize, entry.getTime)
           }
         } finally jar.close()
-      case _ ⇒ None
+      case _ ⇒
+        val conn = url.openConnection()
+        try {
+          conn.setUseCaches(false) // otherwise the JDK will keep the JAR file open when we close!
+          val len = conn.getContentLength
+          val lm = conn.getLastModified
+          Some(ResourceFile(url, len, lm))
+        } catch {
+          case _: Throwable ⇒ None
```

Comment:
Appreciate the input. My caution with your approach is we agree that it's very difficult to test this and the old code worked for quite some time. 

Considering the NPE case, I don't see where the developer or end-user can introduce a URL to the call chain. As best I can tell, they are only generated by the JRE for known good entities. 

That said, is there some manner that your code suggestion could replace what was done in #17495 and generically cover custom URL scheme handlers in the process?